src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IArrayElementReferenceExpression.vb

```diff
@@ -0,0 +1,773 @@
+  Array reference: IParameterReferenceExpression: args (OperationKind.ParameterReferenceExpression, Type: System.String()) (Syntax: 'args')
+  Indices(1):
+      IConversionExpression (Implicit, TryCast: False, Unchecked) (OperatorMethod: Function C.op_Implicit(c As C) As System.Int32) (OperationKind.ConversionExpression, Type: System.Int32) (Syntax: 'c')
+        Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: False, IsUserDefined: True) (MethodSymbol: Function C.op_Implicit(c As C) As System.Int32)
+        Operand: IParameterReferenceExpression: c (OperationKind.ParameterReferenceExpression, Type: C) (Syntax: 'c')
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of InvocationExpressionSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact, WorkItem(22006, "https://github.com/dotnet/roslyn/issues/22006")>
+        Public Sub ArrayElementReference_ExplicitUserDefinedConversionInIndexExpression()
+            Dim source = <![CDATA[
+Option Strict On
+
+Class C
+    Public Sub F(args As String(), c As C)
+        Dim a = args(CType(c, Integer))'BIND:"args(CType(c, Integer))"
+    End Sub
+
+    Public Shared Narrowing Operator CType(ByVal c As C) As Integer
+        Return 0
+    End Operator
+End Class]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IArrayElementReferenceExpression (OperationKind.ArrayElementReferenceExpression, Type: System.String) (Syntax: 'args(CType(c, Integer))')
+  Array reference: IParameterReferenceExpression: args (OperationKind.ParameterReferenceExpression, Type: System.String()) (Syntax: 'args')
+  Indices(1):
+      IConversionExpression (Explicit, TryCast: False, Unchecked) (OperatorMethod: Function C.op_Explicit(c As C) As System.Int32) (OperationKind.ConversionExpression, Type: System.Int32) (Syntax: 'CType(c, Integer)')
+        Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: False, IsUserDefined: True) (MethodSymbol: Function C.op_Explicit(c As C) As System.Int32)
+        Operand: IParameterReferenceExpression: c (OperationKind.ParameterReferenceExpression, Type: C) (Syntax: 'c')
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of InvocationExpressionSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact, WorkItem(22006, "https://github.com/dotnet/roslyn/issues/22006")>
+        Public Sub ArrayElementReference_ExplicitUserDefinedConversionInArrayReference()
+            Dim source = <![CDATA[
+Class C
+    Public Sub F(o As Object, x As Integer)
+        Dim a = DirectCast(o, String())(x)'BIND:"DirectCast(o, String())(x)"
```

Comment:
DirectCast never uses user defined conversion, it looks like scenario differs from the name of the test.

Reply:
Yes, it is a mistake while cloning the test, will rename.

---
In reply to: [141971512](https://github.com/dotnet/roslyn/pull/22410#discussion_r141971512) [](ancestors = 141971512)